Add callout to PrivateLink docs about Environment Variables (#4701)
fishtown-analytics/docs.getdbt.com: website/snippets/_privatelink-hostname-restriction.md

:::caution Environment variables

> Using [Environment variables](/docs/build/environment-variables) when configuring PrivateLink endpoints isn't supported in dbt Cloud. Instead, use [Extended Attributes](/docs/deploy/deploy-environments#extended-attributes) to dynamically change these values in your dbt Cloud environment.

Reviewer: This is only _sorta true,_ since it _technically_ works in ST.  There's some nuance I guess though in that in ST, you dont configure to use a PL endpoint in-app, where as you do in MT.  Do we foresee this causing extra confusion/issues for customers due to this difference?  

To be clear though, I'm more than happy for us to just say across the board, regardless of ST vs MT, that we just dont want folks to use env vars for their connection hostnames if they use PL.

Author: https://dbt-labs.slack.com/archives/C04RSP33V8U/p1704382058007459?thread_ts=1704377784.263199&cid=C04RSP33V8U

> The use of [Environment Variables](https://docs.getdbt.com/docs/build/environment-variables) to define the `hostname` of your PrivateLink endpoint is not supported in dbt Cloud. To dynamically change this value in a given dbt Cloud Environment, you can instead use [Extended Attributes](https://docs.getdbt.com/docs/deploy/deploy-environments#extended-attributes).

Reviewer 2: ```suggestion
Using [Environment variables](/docs/build/environment-variables) to define the `hostname` of your PrivateLink endpoint isn't supported in dbt Cloud. Instead, use [Extended Attributes](/docs/deploy/deploy-environments#extended-attributes) to dynamically change this value in your dbt Cloud environment. 
```
